pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -6,35 +6,49 @@ import (
 	"crypto/x509"
 	"net/http"
 
+	"github.com/pkg/errors"
+
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
 func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.Context, error) {
-	if certificate != "" && key != "" {
-		certKeyClient := &http.Client{}
-		err := GetClientWithCertKey(certKeyClient, certificate, key)
-		if err != nil {
-			return nil, err
-		}
-		return oidc.ClientContext(ctx, certKeyClient), nil
+	certKeyClient, err := GetClientWithCertKey(certificate, key)
+	if err != nil {
+		return nil, err
+	}
+	return oidc.ClientContext(ctx, certKeyClient), nil
+}
+
+func GetClientWithCertKey(certificate, key string) (*http.Client, error) {
+	// Use the default transport to pick up on things like http.ProxyFromEnvironment
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	tlsConfig, err := getTLSClientConfig(certificate, key)
+	if err != nil {
+		return &http.Client{Transport: transport}, err
 	}
-	return ctx, nil
+
+	transport.TLSClientConfig = tlsConfig
+	return &http.Client{Transport: transport}, nil
 }
 
-func GetClientWithCertKey(httpClient *http.Client, certificate, key string) error {
-	if certificate != "" && key != "" {
-		keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
-		if err != nil {
-			return err
-		}
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM([]byte(certificate))
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs:      caCertPool,
-				Certificates: []tls.Certificate{keyPair},
-			},
-		}
+func getTLSClientConfig(certificate, key string) (*tls.Config, error) {
+	if certificate == "" || key == "" {
+		return nil, errors.New("certificate or key provided, but not both")
 	}
```

Comment:
Just realized if cert and key were both empty this function will cause problems. Gonna fix that tonight